react/week-2/lesson.md

```diff
@@ -193,270 +181,297 @@ We don't need to worry about changing the DOM ourselves! This is what makes Reac
+
+### Using `useState`
+
+Now let's look at how we can use the `useState` Hook ([interactive example](https://codesandbox.io/s/initialising-and-rendering-usestate-variables-zgszi?file=/src/Counter.js)):
 
 ```js
-class Counter extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      count: props.count
-    };
-  }
+function Counter() {
+  const [count, setCount] = useState(0);
 
-  render() {
-    // ...
-  }
+  return <p>You clicked {count} times</p>;
 }
 ```
 
-Now the counter component is "remembering" its count, however it is stuck at 0. Next we'll look at how we change what the component is remembering ([interactive example](https://codesandbox.io/s/n714vmyk5l)):
+Let's break this down into small pieces. First, let's look at calling the `useState` function:
 
 ```js
-class Counter extends Component {
-  constructor(props) {
-    // ...
-  }
+useState(0)
+```
 
-  increment = () => {
-    this.setState({
-      count: 1
-    });
-  }
+This initialises the state variable to 0. Any parameter passed to `useState` will be used as the initial value.
 
-  render() {
-    return (
-      <div>
-        Count: {this.state.count}
-        <button onClick={this.increment}>Click me!</button>
-      </div>
-    );
-  }
-}
-```
+Next, let's look at how we render the state variable in our component:
 
-There's a couple of things happening here. We've added an click handler to the button, which will call the `increment` function when the button gets clicked. When the `increment` function is called, it calls `this.setState` function with the `count` set to 1.
+```js
+return <p>You clicked {count} times</p>;
+```
 
-`this.setState` is a special function provided by React's `Component`, and it is used to change what the component is "remembering". It will also tell React that the old value that is still shown in the DOM is outdated and needs to be updated. This will trigger React to re-render, like we did manually with the `renderCounter` function.
+`count` is just a variable, so to insert it into our JSX we treat it like any other variable: we use curly braces.
 
-Now that we have refactored to use React state, we can easily add multiple counters ([interactive example](https://codesandbox.io/s/v8165mq503)):
+Finally, let's at how we get hold of the `count` variable:
 
 ```js
-const App = () => {
-  return (
-    <div>
-      <Counter count={0} />
-      <Counter count={0} />
-      <Counter count={0} />
-    </div>
-  );
-};
+const [count, setCount] = useState(0);
 ```
 
-This still isn't a particular useful application, because we can only still only count to 1! We need to change our `Counter` component so that it reads the previous state, then adds 1 onto that. We can do this by passing a callback function to `this.setState` ([interactive example](https://codesandbox.io/s/qxz27q9y4)):
+To understand this bit of code, we first have to understand *destructuring*.
+
+### Destructuring Arrays
```

Comment:
There's perhaps a discussion to be had about moving this earlier in the syllabus. But I think if we're going to include it, it probably should be in the syllabus (it gets lost otherwise).

Reply:
Actually on second thoughts, I'd be happy to link to it. I'm just not sure where we'd move it to.

Reply:
There's a mention of array destructuring at the bottom of the lesson. Maybe we move it down there and add a link to anchor back to the relevant part of the lesson?

Reply:
Ok yeah. I think as long as we add it to the cheatsheet (for easier reference) that'll be fine.

Reply:
Done in https://github.com/CodeYourFuture/syllabus/pull/448/commits/bb354ca90551b8a6b2cb3a70bb5e3d90b54d4166